htslib/vcf.h

```diff
@@ -999,9 +1085,12 @@ static inline int32_t bcf_dec_int1(const uint8_t *p, int type, uint8_t **q)
     } else if (type == BCF_BT_INT16) {
         *q = (uint8_t*)p + 2;
         return le_to_i16(p);
-    } else {
+    } else if (type == BCF_BT_INT32) {
         *q = (uint8_t*)p + 4;
         return le_to_i32(p);
+    } else { // Invalid type
+        *q = (uint8_t*)p + (1 << bcf_type_shift[(type & 0xf)]);
```

Comment:
It's setting `*q` to match the amount you would expect the pointer to move if you looked it up in `bcf_type_shift`.  It may not be necessary now though.  I think I made this change before modifying `bcf_record_check` to avoid calling `bcf_dec_int1()` for non-integer types.  With that change, it doesn't matter much what happens here.